cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -397,7 +397,12 @@ public Supplier<TwillRunnerService> getTwillRunnerSupplier() {
 
   @Override
   public Collection<MasterEnvironmentTask> getTasks() {
-    return Arrays.asList(podKillerTask, kubeJobCleaner);
+    Set<MasterEnvironmentTask> set = new HashSet<>(2);
+    set.add(kubeJobCleaner);
+    if (podKillerTask != null) {
```

Comment:
Why is this needed? A task shouldn't be null?

Reply:
The podKillerTask is an optional and only initialized if some configurations are explicitly provided.